### PR TITLE
JS: extract YAML library to a shared pack

### PR DESCRIPTION
**Pull request by aibaars**
This pull request moves the YAML library used by JavaScript into a shared QL pack to make it re-usable by other languages.

### CODE REVIEW

File: javascript/ql/lib/semmle/javascript/linters/ESLint.qll [hunk 1]
```diff
@@ -60,22 +60,22 @@ module ESLint {
   }
 
   /** An `.eslintrc.yaml` file. */
-  private class EslintrcYaml extends Configuration, YamlDocument, YamlMapping {
+  private class EslintrcYaml extends Configuration instanceof YamlMapping {
```
Comment (erik-krogh) [inline javascript/ql/lib/semmle/javascript/linters/ESLint.qll:63]:
```suggestion
  private class EslintrcYaml extends Configuration instanceof YamlMapping, YamlDocument {
```

Things might be different if we don't force `EslintrcYaml` to be a `YamlDocument`.

File: shared/yaml/qlpack.yml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+name: codeql/yaml
+version: 0.0.1-dev
+groups: shared
+library: true
+dependencies: 
+  codeql/util: ${workspace}
```
Comment (erik-krogh) [inline shared/yaml/qlpack.yml:6]:
```suggestion
library: true
```

I don't see any imports of `codeql.util`?

**Review by erik-krogh — COMMENTED** [review]
LGTM 👍  

The change should be very non-breaking, as long as you don't do weird `extends` / `@overrides`.  
But it still deserves a chance-note that a new shared YAML library has been added.  

Also, I've written some of this, so I think you should get someone else from dynamic to also approve this PR.

**Comment by erik-krogh** [comment]
And a change-note in `shared/yaml` folder?

**Review by asgerf — DISMISSED** [review]
Also LGTM

**Pull request merged**